embassy-executor/tests/test.rs

```diff
@@ -156,8 +156,7 @@ fn waking_after_completion_does_not_poll() {
     let (executor, trace) = setup();
     executor.spawner().spawn(task1(trace.clone(), waker)).unwrap();
 
-    unsafe { executor.poll() };
```

Comment:
I'm not sure what I was thinking here, the task is single shot and exits by the time this poll returns.